server/controllers/kaban.js

```diff
@@ -0,0 +1,287 @@
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let updatedBoard = await updateBoardBoard({ name }, board_id)
+  res.status(200)
+  res.json({ success: true, board: updatedBoard })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route DELETE /kaban/board
+// @access Protected
+exports.deleteBoard = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  await deleteBoardBoard(board_id);
+  res.status(200)
+  res.json({ success: true })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route POST /column/:board_id
+// @access Protected
+exports.createColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, index } = req.body;
+  const { board_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let column = await createBoardColumn({ name, index }, board_id);
+  res.status(200)
+  res.json({ success: true, column })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /column/:board_id
+// @access Protected
+exports.updateColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { name, index } = req.body;
+  const { board_id, column_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(401)
+   return res.send("Hey!😁, you are not authorized")
+  }
+  let column = await updateBoardColumn({ name, index }, board_id, column_id);
+  res.status(200)
+  res.json({ success: true, column })
+ } catch (error) {
+  res.status(501)
+  throw new Error("Error Processing your request at this time")
+ }
+});
+
+// @route PUT /column/:board_id
+// @access Protected
+exports.deleteColumn = asyncHandler(async (req, res,) => {
+ try {
+  const userId = req.user.id;
+  const { board_id, column_id } = req.params;
+  let board = await Board.findOne({ _id: board_id });
+  if (!board) {
+   res.status(401)
+   return res.send("Column's Board does not exist")
+  }
+  if (board.userId != userId) {
+   res.status(403)
+   return res.send("Hey!😁, you are not authorized")
```

Comment:
This one should be a `401`. Chain these together. No need for a `return` statement as `res.send` already terminates request/response cycle.